_posts/2021-01-11-GPS-serverless.md

```diff
@@ -0,0 +1,195 @@
+
+<center><img src="/img/blog/serverless-GPS/register-device.png" /></center><br/>
+
+[Full tutorial to register a device](https://www.thethingsnetwork.org/docs/devices/registration.html)
+
+After the registration process, I have a tracking device that is connected to the internet and the data is being handled by The Thing Network (TTN).
+
+## Third Step: Create dynamoDB Table as data store
+
+[Amazon DynamoDB](https://docs.aws.amazon.com/dynamodb/) is fully managed NoSQL database service that support key-value and document data structure.
+
+You have the options to create dynamoDB via AWS console , CLI or CloudFormation
+
+If you would like to create a table via cloudformation:
+
+```yaml
+Uplink:
+  Type: AWS::DynamoDB::Table
+  Properties:
+    TableName: "UpLinkTrackerFromLambda"
+    BillingMode: PAY_PER_REQUEST
+    StreamSpecification:
+      StreamViewType: "NEW_IMAGE"
+    AttributeDefinitions:
+      - AttributeName: "id"
+        AttributeType: "S"
+      - AttributeName: "creationTime"
+        AttributeType: "S"
+    KeySchema:
+      - AttributeName: "id"
+        KeyType: "HASH"
+      - AttributeName: "creationTime"
+        KeyType: "RANGE"
+```
+
+## Fourth step: Create a lambda and API Gateway to ingest data from The Thing Network
+
+[AWS Lambda](https://aws.amazon.com/lambda/) is a serverless computing platform that lets you run code without provisioning or managing servers.
+
+I created a simple microservices using API Gateway and lambda.
+
+It is an essential step where allows me to send data between The Thing Network and AWS.
+
+Here is the sample code for lambda written in typescript
+
+```js
+export const saveItemToDynamoDb = async (
+  tableName: string,
+  dynamoDBAttributes: unknown,
+  dynamoDB: DynamoDB
+) => {
+  const attributes: PutItemInputAttributeMap = DynamoDB.Converter.marshall(
+    dynamoDBAttributes
+  );
+
+  const parameter: PutItemInput = {
+    TableName: tableName,
+    Item: attributes,
+  };
+
+  try {
+    const savedItem = await dynamoDB.putItem(parameter).promise();
+    const response = DynamoDB.Converter.unmarshall(savedItem.Attributes);
+    return response;
+  } catch (error) {
+    console.log("There is an error on putting object into table", error);
+  }
+};
+```
+
+The HTTP endpoint will invoke this lambda. Hence, any data from The Thing Network (TTN) will get send into this lambda.
+Those data will be stored in dynamoDB that I created in the previous step.
+
+Once I have the HTTP endpoint, I am ready to integrate AWS and The Thing Network (TTN).
+
+## Fifth Step: Integration between The Thing Network (TTN) and AWS
+
+Integration method available in The Thing Network (TTN):
+
+<center><img src="/img/blog/serverless-GPS/integrationMethod.png" /></center><br/>
+
+In order to have it integrate with the http endpoint that I created in previous step, I will go with HTTP integration method here.
+
+## Sixth Step: Create the application - React App
+
+**Before I start**
```

Comment:
**Before I start**

Figure out if it is YOU or them

Reply:
I usually write `we` since I'm intending it to be a conversation between me and the reader.